src/v3/src/transformer/i18n/transform.ts

```diff
@@ -73,6 +83,65 @@ const updateLabel = (transaction: IdxTransaction, element: FieldElement): void =
   }
 };
 
+const transformCustomizedErrorKeys = (
+  transaction: IdxTransaction,
+  formBag: FormBag,
+  widgetProps: WidgetProps,
+): void => {
+  const { nextStep: { name } = {} } = transaction;
+  if (name !== IDX_STEP.IDENTIFY) {
+    return;
+  }
+
+  const { dataSchema, uischema: { elements } } = formBag;
+  [
+    { field: 'identifier', key: 'error.username.required' },
+    { field: 'credentials.passcode', key: 'error.password.required' },
+  ].forEach((obj) => {
+    const ele = getUIElementWithName(
+      obj.field,
+      elements as UISchemaElement[],
+    ) as FieldElement;
+    if (ele && isCustomizedI18nKey(obj.key, widgetProps)) {
+      // eslint-disable-next-line no-param-reassign
+      dataSchema[obj.field] = {
+        validate(data) {
+          const isValid = !!data[obj.field];
+          return isValid ? undefined : [{
+            message: loc(obj.key, 'login'),
+            i18n: { key: obj.key },
+          }];
+        },
+      };
+    }
+  });
+};
+
+const transformElementSubLabels = (
```

Comment:
since you are touching logic here, can we move directly to `translations[]` approach instead of keep adding custom field in schema? if it's too much effort, we can start with identify form first

Reply:
centralized translation step does not mean manage each custom translation in one file, but figure out a general pattern that can be applied to most/all custom steps.